### DISCUSSION TITLE
CI is failing, can it be silenced

### DISCUSSION THREAD

**Discussion by cool-RR** [saulpw/visidata#1814] [General]
CI seems to be broken. When I make a PR I get a failure email. This happens to many projects. I think it's annoying because it makes people get used to failure emails and disregard them.

Since we don't know if and when this'll be fixed, is it possible to just not show the failures or not send emails for them or something? (I wouldn't want to disable GitHub email notifications because some projects have a working CI.)

**Comment by frosencrantz** [comment]
It looks like the tests are passing except for Python 3.10. Anja directly keeps the CI running well and resolves breakages.  It is hard for me to read the cmdlog output.  It says something failed, but I can't tell what failed from the output.  It would be nice if the output contained more details about which test was run, and if a test failed, which test failed.  That may be there, but I'm not sure how to read it.

I am not directly adding code to VisiData, but I like to advocate for its health.  I worry if the notifications are not being sent, the failures would be more likely to be ignored or missed.
 
Would it be preferable to make the tests pass  rather than turn off the notifications?

**Reply by cool-RR** [reply]
My strong opinion from bitter experience is that the goal of "Make people not be frustrated by fake errors" shouldn't depend at all on the goal of "Make the tests pass".

**Reply by anjakefala** [reply]
The tests are currently failing randomly because of this bug: https://github.com/saulpw/visidata/issues/1808. It does not have a clear stacktrace, because it basically gets stuck on the `requests` import. I added a feature that aborts the CI run, if it goes on longer than 15 minutes, otherwise it would eat up all of our minutes. It is under active investigation.

**Reply by anjakefala** [reply]
> is it possible to just not show the failures or not send emails for them or something?

I have completely turned off all GitHub emails for my account, so I fully understand that they can be frustrating. This is something you can explore controlling with your email preferences, and not something under our fine-tuned control unless we completely turn off the CI, which we are not going to for a bug that we are actively working on.

 I believe you can play with your respository watch settings: https://docs.github.com/en/account-and-profile/managing-subscriptions-and-notifications-on-github/setting-up-notifications/configuring-notifications#about-participating-and-watching-notifications

This might also be helpful: https://docs.github.com/en/account-and-profile/managing-subscriptions-and-notifications-on-github/setting-up-notifications/configuring-notifications#customizing-your-email-notifications

You can also absolutely add an email filter for VisiData CI so it skips your inbox. 

I have paused the `vdsql` tests until we have time to invest in them. But have kept the one for the `requests` import on. We need confirmation on which Python versions it tends to affect.

**Comment by frosencrantz** [comment]
Happy to see the CI is back to passing!  😄   Wondering if it would be useful when there is a long running failure that cannot be fixed directly in VisiData, if that test could be moved to a separate test run like a "hospital" run.  A failing test would stay in the hospital run, and it would fail regularly in that run, once the issue is passing again it could be discharged from the hospital and go back to the main run. 

It would allow the regular run to pass more often to help identify new failures.

**Reply by anjakefala** [reply]
Hmmm. This is not the worst suggestion....

**Reply by anjakefala** [reply]
Why does github not provide :thinking: as an emoji react for comments :((
